File: LeetCode/150_Evaluate_Reverse_Polish_Notation.py
def evalRPN(tokens):
    stack = []
    for ops in tokens:
        if ops == "+":
            stack.append(stack.pop() + stack.pop())
        elif ops == "-":
            a, b = stack.pop(), stack.pop()
            stack.append(b - a)
        elif ops == "*": 
            stack.append(stack.pop() * stack.pop())
        elif ops == "/":
            a, b = stack.pop(), stack.pop()
            stack.append(int(b / a))
        else:
            stack.append(int(ops))
    
    return stack[0]



tokens = ["2","1","+","3","*"]
Output = 9
tokens = ["4","13","5","/","+"]
Outpu = 6

evalRPN(tokens)

# Time Complexity: O(n)
# 

File: LeetCode/150_Evaluate_Reverse_Polish_Notation.js
var evalRPN = function(tokens) {
    var stack = [];
    for (let i in tokens) {
        if (tokens[i] == "+") {
            stack.push(stack.pop() + stack.pop())
        } else if (tokens[i] == "-") {
            let a = stack.pop()
            let b = stack.pop()
            stack.push(b - a)
        } else if (tokens[i] == "*") {
            stack.push(stack.pop() * stack.pop())
        } else if (tokens[i] == "/") {
            let a = stack.pop()
            let b = stack.pop()
            stack.push(parseInt(b/a))
        } else {
            stack.push(parseInt(tokens[i]))
        }
    }
    return stack[0]
}